python/lsst/ctrl/oods/gen3ButlerIngester.py

```diff
@@ -263,11 +263,18 @@ def clean(self):
         # get all TAGGED collections
         tagged_cols = list(butler.registry.queryCollections(collectionTypes=CollectionType.TAGGED))
 
-        # get all TAGGED datasets
-        tagged_datasets = set(butler.registry.queryDatasets(datasetType=..., collections=tagged_cols))
-
-        # get a set of datasets in all_datasets, but not in tagged_datasets
-        ref = all_datasets.difference(tagged_datasets)
+        # Note: The code below is to get around an issue where passing
+        # an empty list as the collections argument to queryDatasets
+        # returns all datasets.
+        if len(tagged_cols) == 0:
```

Comment:
Works for me, but some Python purists may frown upon this as PEP 8 (and, by extension, [LSST dev guide](https://developer.lsst.io/python/style.html#comparisons)) recommends using the fact that empty sequences are false.

Reply:
Fixed